src/Mailer/Transport/SmtpTransport.php

```diff
@@ -465,7 +465,7 @@ protected function _smtpSend(?string $data, $checkCode = '250'): ?string
                 $response .= $bytes;
             }
             if (substr($response, -2) !== "\r\n") {
-                throw new SocketException('SMTP timeout.');
+                throw new SocketException($response ?: 'SMTP timeout.');
```

Comment:
Won't `response` always be not null as it is initialized to `''` on line 458.

Reply:
This checks for a falsy or empty string this way.  I think maybe it should be more explict though.

Reply:
With an `IF` statement vs. the short way? 

I didn't want to add another nesting level already had `while` and `if` but if you think its better for readability, I can do so.

Reply:
No, I was just thinking how to explain what's happening. A comment would do well to explain why timeout is the default and why response is sent if not empty. Looks fine to me technically.

Reply:
My bad I conflated ?? And ?:.